src/extension/src/file_handlers/ExtConfigSettingsHandler.py

```diff
@@ -33,37 +33,70 @@ def __init__(self, logger, json_file_handler, config_folder):
         self.public_settings_key = Constants.PUBLIC_SETTINGS
         self.public_settings_all_keys = Constants.ConfigPublicSettingsFields
 
-    def get_seq_no(self):
+    def get_seq_no(self, is_enable_request=False):
         """ Fetches sequence number, initially from the env variable. If nothing is set in env variable then fetches from config folder based on timestamp, since GA updates the settings file before calling a command """
         try:
-            seq_no = os.getenv(Constants.SEQ_NO_ENVIRONMENT_VAR)
-            if seq_no is not None:
+            # get seq no from env var
+            seq_no = self.__get_seq_no_from_env_var()
+            if seq_no == -1:
+                self.logger.log_error("Error occurred while verifying sequence number between environment variable and configuration settings")
+                return None
```

Comment:
__get_seq_no_from_env_var is doing more than its name suggests (the name is good, but breaks the single responsibility principle). The check to see if the seq no. exists on the file system after being discovered there should happen outside that function. get_seq_no should take on additional responsibility appropriately.